test-executor/src/main/java/com/cognifide/aet/executor/SuiteRunner.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Automated Exploratory Tests
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.aet.executor;
+
+import com.cognifide.aet.communication.api.messages.MessageType;
+import com.cognifide.aet.communication.api.messages.TaskMessage;
+import com.cognifide.aet.communication.api.metadata.Suite;
+import com.cognifide.aet.communication.api.execution.ProcessingStatus;
+import com.cognifide.aet.communication.api.execution.SuiteStatusResult;
+import com.cognifide.aet.executor.common.MessageProcessor;
+import com.cognifide.aet.executor.common.ProcessorFactory;
+import com.cognifide.aet.executor.common.RunnerTerminator;
+import com.cognifide.aet.rest.LockService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.ObjectMessage;
+import javax.jms.Session;
+
+/**
+ * This class is responsible for running a single test suite and checking the processing status.
+ */
+public class SuiteRunner implements Runnable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SuiteRunner.class);
+
+  private Session session;
+
+  private MessageProducer messageProducer;
+
+  private MessageConsumer messageConsumer;
+
+  private LockService lockService;
+
+  private RunnerCacheUpdater runnerCacheUpdater;
+
+  private SuiteStatusHandler suiteStatusHandler;
+
+  private RunnerTerminator runnerTerminator;
+
+  private Suite suite;
+
+  private String inQueueName;
+
+  private long messageReceiveTimeout;
+
+  public SuiteRunner(Session session, LockService lockService, RunnerCacheUpdater runnerCacheUpdater,
+      SuiteStatusHandler suiteStatusHandler, Suite suite, String inQueueName, long messageReceiveTimeout) {
+    this.session = session;
+    this.lockService = lockService;
+    this.runnerCacheUpdater = runnerCacheUpdater;
+    this.suiteStatusHandler = suiteStatusHandler;
+    this.runnerTerminator = new RunnerTerminator();
+    this.suite = suite;
+    this.inQueueName = inQueueName;
+    this.messageReceiveTimeout = messageReceiveTimeout;
+  }
+
+  /**
+   * Runs the test suite processing and starts a thread which checks the processing status.
+   *
+   * @throws JMSException
+   */
+  public void runSuite() throws JMSException {
```

Comment:
I dream of using [RxJava](https://github.com/ReactiveX/RxJava) here in the future.